src/cljs/sixsq/slipstream/webui/authn/views.cljs

```diff
@@ -180,10 +194,74 @@
         (case modal-kw
           :login [:span (@tr [:no-account?]) " "
                   [:a {:on-click f :style {:cursor "pointer"}} (str (@tr [:signup-link]))]]
+          :reset-password [:span (@tr [:already-registered?]) " "
+                           [:a {:on-click f :style {:cursor "pointer"}} (str (@tr [:login-link]))]]
           :signup [:span (@tr [:already-registered?]) " "
                    [:a {:on-click f :style {:cursor "pointer"}} (str (@tr [:login-link]))]])))))
 
 
+(defn reset-password-modal
+  "Modal that holds the authentication (login or sign up) forms."
+  []
+  (let [open-modal (subscribe [::authn-subs/open-modal])
+       form-id (subscribe [::authn-subs/form-id])
+       cep (subscribe [::cimi-subs/cloud-entry-point])
+       username (subscribe [::authn-subs/username])
+       error-message (subscribe [::authn-subs/error-message])
+       success-message (subscribe [::authn-subs/success-message])
+       {:keys [baseURI]} @cep
+       form-id "reset-password-modal-form"
+       tr (subscribe [::i18n-subs/tr])]
+
+    [ui/Modal
+      {:id        "reset-password-modal"
+       :size      :tiny
+       :open      (= @open-modal :reset-password)
+       :closeIcon true
+       :on-close  (fn []
+                    (dispatch [::authn-events/close-modal])
+                    (dispatch [::authn-events/clear-success-message])
+                    (dispatch [::authn-events/clear-error-message]))}
+
+      [ui/ModalHeader (@tr [:reset-password])]
+
+      [ui/ModalContent
+
+        (when @error-message
+          [ui/Message {:negative  true
+                      :size      "tiny"
+                      :onDismiss #(dispatch [::authn-events/clear-error-message])}
+          [ui/MessageHeader (@tr [:reset-password-error])]
+          [:p @error-message]])
+
+        (when @success-message
+          [ui/Message {:negative  false
+                      :size      "tiny"
+                      :onDismiss #(dispatch [::authn-events/clear-success-message])}
+          [ui/MessageHeader (@tr [:reset-password-success])]
+          [:p @success-message]])
+
+        [ui/FormInput {:name         "username"
+                      :type         "text"
+                      :placeholder  "Username"
+                      :icon         "user"
+                      :iconPosition "left"
+                      :required     true
+                      :autoComplete "off"
+                      :on-change #(dispatch [::authn-events/username (-> % .-target .-value)])}]
```

Comment:
There's a callback utility to handle callback functions like this in `sixsq.slipstream.webui.utils.ui-callback` namespace.

Does the username need to be stored in the database at all?  Wouldn't a local atom work?